src/highlight/HighlightManager.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import BaseManager, { Options, Props } from '../common/BaseManager';
+import HighlightContainer from './HighlightContainer';
+
+export default class HighlightManager implements BaseManager {
+    location: number;
+
+    reactEl: HTMLElement;
+
+    constructor({ location = 1, referenceEl }: Options) {
+        this.location = location;
+        this.reactEl = this.insert(referenceEl);
+    }
+
+    destroy(): void {
+        ReactDOM.unmountComponentAtNode(this.reactEl);
+
+        this.reactEl.remove();
+    }
+
+    exists(parentEl: HTMLElement): boolean {
+        return parentEl.contains(this.reactEl);
+    }
+
+    insert(referenceEl: HTMLElement): HTMLElement {
+        // Find the nearest applicable reference and document elements
+        const documentEl = referenceEl.ownerDocument || document;
+        const parentEl = referenceEl.parentNode || documentEl;
+
+        // Construct a layer element where we can inject a root React component
+        const rootLayerEl = documentEl.createElement('div');
+        rootLayerEl.classList.add('ba-Layer');
```

Comment:
Do we have any concerns about rendering this layer even though the feature is not yet available?

Reply:
If the toolbar highlight button is not clicked, we actually only render a `<div />` with absolute positioning and `pointer-event: none`. So I don't have any concerns.

Reply:
Gotcha. I think @ConradJChan had some concerns about rendering these components unconditionally, but I don't recall why. It should be okay for now, but we'll need to keep this in mind if we add anything else outside of the "creator" components.